modules/gdscript/editor/gdscript_docgen.cpp

```diff
@@ -191,48 +253,39 @@ void GDScriptDocGen::generate_docs(GDScript *p_script, const GDP::ClassNode *p_c
 				p_script->member_lines[var_name] = m_var->start_line;
 
 				DocData::PropertyDoc prop_doc;
-
 				prop_doc.name = var_name;
 				prop_doc.description = m_var->doc_data.description;
 				prop_doc.is_deprecated = m_var->doc_data.is_deprecated;
 				prop_doc.is_experimental = m_var->doc_data.is_experimental;
+				_doctype_from_gdtype(m_var->get_datatype(), prop_doc.type, prop_doc.enumeration);
 
-				GDType dt = m_var->get_datatype();
-				switch (dt.kind) {
-					case GDType::CLASS:
-						prop_doc.type = _get_class_name(*dt.class_type);
-						break;
-					case GDType::VARIANT:
-						prop_doc.type = "Variant";
-						break;
-					case GDType::ENUM:
-						prop_doc.type = Variant::get_type_name(dt.builtin_type);
-						// Replace :: from enum's use of fully qualified class names with regular .
-						prop_doc.enumeration = String(dt.native_type).replace("::", ".");
-						break;
-					case GDType::NATIVE:;
-						prop_doc.type = dt.native_type;
+				switch (m_var->property) {
+					case GDP::VariableNode::PROP_NONE:
 						break;
-					case GDType::BUILTIN:
-						prop_doc.type = Variant::get_type_name(dt.builtin_type);
+					case GDP::VariableNode::PROP_INLINE:
+						if (m_var->setter != nullptr) {
+							prop_doc.setter = m_var->setter->identifier->name;
+						}
+						if (m_var->getter != nullptr) {
+							prop_doc.getter = m_var->getter->identifier->name;
+						}
 						break;
-					default:
-						// SCRIPT: can be preload()'d and perhaps used as types directly?
-						// RESOLVING & UNRESOLVED should never happen since docgen requires analyzing w/o errors
+					case GDP::VariableNode::PROP_SETGET:
+						if (m_var->setter_pointer != nullptr) {
+							prop_doc.setter = m_var->setter_pointer->name;
+						}
+						if (m_var->getter_pointer != nullptr) {
+							prop_doc.getter = m_var->getter_pointer->name;
+						}
 						break;
 				}
 
-				if (m_var->property == GDP::VariableNode::PROP_SETGET) {
-					if (m_var->setter_pointer != nullptr) {
-						prop_doc.setter = m_var->setter_pointer->name;
+				if (m_var->initializer) {
+					if (m_var->initializer->is_constant) {
+						prop_doc.default_value = m_var->initializer->reduced_value.get_construct_string().replace("\n", "\\n");
+					} else {
+						prop_doc.default_value = "<unknown>";
```

Comment:
I just copied this from function signature hint. If we want to change this, then we need to do it here as well:

https://github.com/godotengine/godot/blob/5444afae63f37cb4b890c781e5679a3b7ee5a392/modules/gdscript/gdscript_editor.cpp#L720-L722

More options: `<non-constant>`, `<non-const-expr>`, `<dynamic>`, `<unresolved>`.

Reply:
I personally really like `<non-const-exp>` for some reason! I just feel like it's worth explaining _why_ something is unknown, as opposed to just stating that it is. That way people can more easily and immediately understand what's happening.

I don't think changing `gdscript_editor.cpp` would lead to significant compat breaking, right?

Reply:
It's a bit ugly in my opinion because of the two hyphens and longer text. I would choose `<unknown>`, `<dynamic>` or `<non-constant>`.

Reply:
I see your own, and my opinion is certainly very subjective. I don't think `<unknown>` captures the reason why a value can't be written. I think `<dynamic>` hints at it but isn't the most explicit, whereas `<non-constant>` is the clearest to me, though I agree it isn't the prettiest.

Reply:
Let's keep `<unknown>` for now since we haven't reached a consensus and there is a precedent with signature hint. We can fix this later or maybe some other reviewer will give their opinion.

Reply:
Ok, sounds good! 👍